tests/integration/test_http_get_all.py

```diff
@@ -0,0 +1,100 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import utils
+
+
+def test_http_get_all(test_env):
+    """
+    Check the API endpoint to get all requests.
+
+    Process:
+    * Pre-populate data in Cachito by submitting basic requests and waiting for them to complete.
+    * Verify the submitted requests are not incorrectly marked as "in_progress".
+    * Verify the submitted requests can be found by using the state=complete filter.
+    Checks:
+    * Check that response code is 200
+    * Check that filtering parameters are working properly (page, per_page, and state)
+    """
+    client = utils.Client(test_env["api_url"], test_env["api_auth_type"])
+    requests_amount = test_env["http_get_all"]["requests_amount"]
+    submitted_requests = []
+    payload = {
+        "repo": test_env["package"]["repo"],
+        "ref": test_env["package"]["ref"],
+        "pkg_managers": test_env["package"]["pkg_managers"],
+    }
+    initial_responses = [client.create_new_request(payload) for _ in range(requests_amount)]
+    for initial_response in initial_responses:
+        completed_response = client.wait_for_complete_request(initial_response)
+        response_data = completed_response.data
+        submitted_requests.append(response_data["id"])
+
+    assert_no_requests_in_progress_state(client, requests_amount, submitted_requests)
+    assert_completed_requests(client, submitted_requests)
+
+
+def test_get_all_verbose(test_env):
+    """
+    Check the API endpoint to get all requests with the verbose flag enabled.
+
+    Process:
+    * Create a request and wait for it to complete
+    * Send request get_all to the Cachito API with verbose
+    Checks:
+    * Check that response code is 200
+    * Check that verbose is working properly
+    """
+    client = utils.Client(test_env["api_url"], test_env["api_auth_type"])
+    initial_response = client.create_new_request(
+        payload={
+            "repo": test_env["package"]["repo"],
+            "ref": test_env["package"]["ref"],
+            "pkg_managers": test_env["package"]["pkg_managers"],
+        },
+    )
+    client.wait_for_complete_request(initial_response)
+
+    query_params = {"per_page": 1, "page": 1, "state": "complete", "verbose": True}
+    request_id = initial_response.id
+    found_request = False
+    while not found_request:
+        response = client.fetch_all_requests(query_params, all_pages=False)
+        if response.status != 200:
+            break
+        if response.data["items"][0]["id"] == request_id:
+            found_request = True
```

Comment:
Optional: If you prefer, you can not have the `found_request` variable by doing `while True:` and then replacing this specific line with `break`.

Reply:
Thank you, I thought about it. But I think it is a good idea trying to avoid while true loop in non-one-ff code.
I would like to save it as it is.